### PR TITLE
feat: Add a Release Checks Template

### PR DESCRIPTION
**Pull request by eugene-davis**
Restructures the python pipeline to do the following:
1. Always build the artifacts to be published (and if docker is included, build the image as well)
2. Reduce the number of times a "condition" must be set on an individual step. This has increased the number of files as a result, but I think makes it easier to follow what is going on.

### CODE REVIEW

File: jobs/python.setup.py.yml [hunk 1]
```diff
@@ -61,13 +61,26 @@ jobs:
   steps:
   - template: ../steps/python/python.build.yml
 
+jobs:
```
Comment (martinm82) [inline jobs/python.setup.py.yml]: This is already defined above

File: jobs/python.setup.py.yml [hunk 2]
```diff
@@ -61,13 +61,26 @@ jobs:
...
+- job: 'ReleaseChecks'
+  dependsOn:
+    - macOS
+    - Linux
+    - Windows
+  pool:
+    vmImage: 'ubuntu-16.04'
+  condition: and(succeeded('macOS'), succeeded('Windows'), succeeded('Linux'), succeeded('ReleaseChecks'), 'PullRequest')
```
Comment (martinm82) [inline jobs/python.setup.py.yml]: The check for a PR should be like this `eq(variables['build.reason'], 'PullRequest')`

Comment (martinm82) [inline jobs/python.setup.py.yml]: A condition on itself? Looks like a copy&paste error

File: steps/python/python.release.checks.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+- script: |
+    pip install --no-cache-dir \
+      bumpversion==0.5.3 \
+      twine==1.12.1 \
+      setuptools==40.8.0 \
+      pip==19.0.3
+  displayName: "Install publish prerequisites"
+
+- script: "bumpversion --dry-run --message 'Bump version: {current_version} -> {new_version} ***NO_CI***' release"
```
Comment (martinm82) [inline steps/python/python.release.checks.yml]: Could we instead add this "check" feature to the existing release template?

**Review by martinm82 — CHANGES_REQUESTED** [review]

File: steps/python/python.release.checks.yml (hunk 3 above)
Comment (eugene-davis) [reply steps/python/python.release.checks.yml]: Maybe, right now I'm playing with it (provided the draft to get initial feedback like this). Once I know what all will go into it, I will revisit it.

**Comment by eugene-davis** [comment]
See execution here: https://dev.azure.com/tomtomweb/GitHub-TomTom-International/_build/results?buildId=2832

**Review by ghost — APPROVED** [review]

**Comment by martinm82** [comment]
Closing due to inactivity

**Pull request closed without merging**